datastream-utils/src/main/java/com/linkedin/datastream/common/LogUtils.java

```diff
@@ -62,4 +66,30 @@ public static String logNumberArrayInRange(List<Integer> list) {
       return list.toString();
     }
   }
+
+  /**
+   * Shortening the list of topic-partition mappings by merging partitions of the same topic together. e.g.
+   * topic1-0, topic1-1, topic2-0 -> topic1:[0-1], topic2:[0]
+   * @param list list of strings to generate logging String for
+   * @return compacted String that merges partitions per topic
+   */
```

Comment:
Suggestion: Name of the method can change to `logSummarizedTopicPartitionsMapping`